CONTRIBUTING.md

```diff
@@ -66,6 +66,20 @@ Ensure you have the following installed:
    ```
    export ASTRO_API_TOKEN=<your-api-token>
    ```
+   
+## Adding Dependencies
+
+This provider uses [Go modules](https://go.dev/wiki/Modules).
+Please see the Go documentation for the most up-to-date information about using Go modules.
+
+To add a new dependency `github.com/author/dependency` to your Terraform provider:
+
+```shell
+go get github.com/author/dependency
+go mod tidy
+```
+
+Then commit the changes to `go.mod` and `go.sum`.
 
 ## Making Changes
```

Comment:
note for myself - i should probably add a section that breaks down the directory structure more here so it's clearer what is what and where a developer should start